src/Elements/Geometry/Ray.cs

```diff
@@ -77,6 +78,102 @@ public bool Intersects(Triangle tri, out Vector3 result)
             }
         }
 
+        /// <summary>
+        /// Does this ray intersect with the provided SolidOperation?
+        /// </summary>
+        /// <param name="solidOp">The SolidOperation to intersect with.</param>
+        /// <param name="result">The intersection result.</param>
+        /// <returns>True if an intersection occurs, otherwise false. If true, check the intersection result for the location of the intersection.</returns>
+        public bool Intersects(SolidOperation solidOp, out Vector3 result)
```

Comment:
Can we make the signature `Intersects(SolidOperation solidOp, out List<Vector3> result)`? Then we can record all intersections along the ray.

Reply:
Done.